core/src/SYCL/Kokkos_SYCL_Parallel_Team.hpp

```diff
@@ -129,30 +127,34 @@ class TeamPolicyInternal<Kokkos::Experimental::SYCL, Properties...>
   }
   inline bool impl_auto_vector_length() const { return m_tune_vector_length; }
   inline bool impl_auto_team_size() const { return m_tune_team_size; }
+  // FIXME_SYCL This is correct in most cases, but not necessarily in case a
+  // custom sycl::queue is used to initialize the execution space.
   static int vector_length_max() {
-    // FIXME_SYCL provide a reasonable value
-    return 1;
+    std::vector<size_t> sub_group_sizes =
+        execution_space{}
+            .impl_internal_space_instance()
+            ->m_queue->get_device()
+            .template get_info<sycl::info::device::sub_group_sizes>();
+    return *std::max_element(sub_group_sizes.begin(), sub_group_sizes.end());
   }
 
-  static int verify_requested_vector_length(int requested_vector_length) {
+ private:
+  int verify_requested_vector_length(int requested_vector_length) const {
```

Comment:
can be static with your recent changes

Reply:
Yes, it can be at the moment. Do you prefer I change this again?

Reply:
Yes

Reply:
Fixed.